ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/LedgerDB/OrphanArbitrary.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE TypeApplications #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+module Test.Ouroboros.Storage.LedgerDB.OrphanArbitrary where
+
+import           Test.QuickCheck
+import           Data.Time.Clock (DiffTime, secondsToDiffTime, diffTimeToPicoseconds)
+import           Ouroboros.Consensus.Config.SecurityParam (SecurityParam(..))
+import           Ouroboros.Consensus.Storage.LedgerDB.DiskPolicy (RequestedSnapshotInterval(..))
+
+{-------------------------------------------------------------------------------
+  Orphan Arbitrary instances
+-------------------------------------------------------------------------------}
+
+instance Arbitrary SecurityParam where
+  arbitrary = SecurityParam <$> choose (0, 6)
+  shrink (SecurityParam k) = SecurityParam <$> shrink k
+
+instance Arbitrary RequestedSnapshotInterval where
+  arbitrary =
+    oneof [ requestedSnapshotInterval
+          , defaultSnapshotInterval
+          ]
+  shrink DefaultSnapshotInterval = []
+  shrink (RequestedSnapshotInterval v) = RequestedSnapshotInterval <$> shrink v
+
+requestedSnapshotInterval :: Gen RequestedSnapshotInterval
+requestedSnapshotInterval =
+  RequestedSnapshotInterval <$> arbitrary
+
+defaultSnapshotInterval :: Gen RequestedSnapshotInterval
+defaultSnapshotInterval = pure DefaultSnapshotInterval
+
+instance Arbitrary DiffTime where
+  arbitrary = fromIntegral @Int <$> arbitrary
```

Comment:
I have to check the details every time, but the `Arbitrary Int` instance is often _not_ what people would expect; it's a bit of a notorious wart. Therefore, I generally call `choose` explicitly.